components/brave_rewards/resources/ui/components/adsBox.tsx

```diff
@@ -140,39 +286,58 @@ class AdsBox extends React.Component<Props, State> {
     const toggle = !(!enabledMain || !adsUIEnabled || !adsIsSupported)
     const showDisabled = firstLoad !== false || !toggle || !adsEnabled || !adsIsSupported
 
+    if (!adsHistory) {
+      return null
+    }
+
+    const rows = this.getAdHistoryData(adsHistory, savedOnly)
+
     return (
-      <Box
-        title={getLocale('adsTitle')}
-        type={'ads'}
-        description={getLocale('adsDesc')}
-        toggle={toggle}
-        checked={enabled}
-        settingsChild={this.adsSettings(enabled && enabledMain)}
-        testId={'braveAdsSettings'}
-        disabledContent={showDisabled ? this.adsDisabled() : null}
-        onToggle={this.onAdsSettingChange.bind(this, 'adsEnabled', '')}
-        settingsOpened={this.state.settings}
-        onSettingsClick={this.onSettingsToggle}
-        attachedAlert={this.adsNotSupportedAlert(adsIsSupported)}
-      >
-        <List title={getLocale('adsCurrentEarnings')}>
-          <Tokens
-            value={estimatedPendingRewards}
-            converted={utils.convertBalance(estimatedPendingRewards, balance.rates)}
-          />
-        </List>
-        <List title={getLocale('adsPaymentDate')}>
-          <NextContribution>
-            {nextPaymentDate}
-          </NextContribution>
-        </List>
-        <List title={getLocale('adsNotificationsReceived')}>
-          <Tokens
-            value={adNotificationsReceivedThisMonth.toString()}
-            hideText={true}
+      <>
+        <Box
+          title={getLocale('adsTitle')}
+          type={'ads'}
+          description={getLocale('adsDesc')}
+          toggle={toggle}
+          checked={enabled}
+          settingsChild={this.adsSettings(enabled && enabledMain)}
+          testId={'braveAdsSettings'}
+          disabledContent={showDisabled ? this.adsDisabled() : null}
+          onToggle={this.onAdsSettingChange.bind(this, 'adsEnabled', '')}
+          settingsOpened={this.state.settings}
+          onSettingsClick={this.onSettingsToggle}
+          attachedAlert={this.adsNotSupportedAlert(adsIsSupported)}
+        >
+          <List title={getLocale('adsCurrentEarnings')}>
+            <Tokens
+              value={estimatedPendingRewards}
+              converted={utils.convertBalance(estimatedPendingRewards, balance.rates)}
+            />
+          </List>
+          <List title={getLocale('adsPaymentDate')}>
+            <NextContribution>
+              {nextPaymentDate}
+            </NextContribution>
+          </List>
+          <List title={getLocale('adsNotificationsReceived')}>
+            <Tokens
+              value={adNotificationsReceivedThisMonth.toString()}
+              hideText={true}
+            />
+          </List>
+          <ShowAdsHistory onAdsHistoryOpen={this.onAdsHistoryToggle} notEmpty={rows && rows.length !== 0}/>
```

Comment:
probably we want to show this only when you have some history

Reply:
Waiting for feedback from @jenn-rhim on this, I think it's the same as above...